pkg/model/serializer.go

```diff
@@ -761,3 +830,25 @@ func (d *Duration) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	*d = Duration(duration)
 	return nil
 }
+
+// UnmarshalYAML Implements the Unmarshaler interface of the yaml pkg.
+func (a *Affinity) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
Maybe a comment explaining why we unmarshal the affinity, marshal it again in json to unmarshalling again would help future maintainability of this code